Sources/Sentry/SentryScope.m

```diff
@@ -319,6 +326,33 @@ - (void)addAttachment:(SentryAttachment *)attachment
     }
 }
 
+- (void)setTransaction:(id<SentrySpan>)transaction
+{
+    self.currentTransaction = transaction;
+    [self notifyListeners];
```

Comment:
I wonder if we should notify listeners and trigger scope serialization here.
This is for persistance of data on hard crashes only.

In this case I think it's best not to trigger it:

```suggestion
```

Reply:
it should not, same thing for `serialize`